istio/src/main/scala/io/buoyant/k8s/istio/ClusterCache.scala

```diff
@@ -3,8 +3,13 @@ package io.buoyant.k8s.istio
 import com.twitter.logging.Logger
 import com.twitter.util.{Closable, Future, Time}
 
-class ClusterCache(client: DiscoveryClient) extends Closable {
-  import ClusterCache._
+case class Cluster(dest: String, port: String)
+
+trait ClusterCache extends Closable {
+  def get(domain: String): Future[Option[Cluster]]
+}
+
+class ClusterCacheBackedByApi(client: DiscoveryClient) extends ClusterCache {
```

Comment:
This feels a bit verbose.  How do you feel about something like `ApiClusterCache`?

Reply:
I tend to prefer more descriptive names, but I am happy with the suggestion. The only thing I'd suggest is that we keep the same naming pattern around things, do we usually go with `Qualifier` + `Thing`, as in `Api` `ClusterCache` or do we go the other way around? Looking around the code, it seems like so far we have relied on packages to namespace different implementations but keep the same name, should I do this here?

Reply:
third option: `ClusterCache with Api` (Alex will _hate_ this :D)

Reply:
:P I actually find the cake pattern super cute, but had way too many problems debugging badly wired software 😤